xbmc/CompileInfo.h

```diff
@@ -18,8 +18,9 @@ class CCompileInfo
   static const char* GetPackage();
   static const char* GetClass();
   static const char* GetAppName();
-  static const char *GetSuffix();  // Git "Tag", e.g. alpha1
+  static const char* GetSuffix();  // Git "Tag", e.g. alpha1
   static const char* GetSCMID();   // Git Revision
+  static std::string GetTrustedAddonRepos();
```

Comment:
@yol i first tried with `static const std::string& GetTrustedAddonRepos();` here and that built fine on stretch, but caused a crash on startup (crashlog) is that a typical pitfall 🤔 ? i wouldn‘t notice if i hadn‘t runtime-tested

Reply:
Uh yes it is quite typical and I feel ashamed for saying you could use `const std::string&` in this situation; I clearly wasn't paying enough attention. You are constructing the string inside the function as a local variable. Then you return a reference to that variable. When you leave the function, the string gets destructed because it was local. Now you have a dangling reference that you cannot use. The compiler should give you a warning, though :shrug:

Reply:
```
a.cpp:3:33: warning: returning reference to local temporary object [-Wreturn-stack-address]
const std::string& a() { return "a"; }
                                ^~~
```

That's what I get

Reply:
ahhh, gotcha. no reason to be ashamed. i might have missed the warning, as i usually hack in -> build -> test :) .. thanks for clarifying